WordPress/gutenberg
# Zoom Out: Disable zooming out when Distraction Free mode is activated

Fixes: #67025 

## What?

The Zoom Out and Distraction Free Mode features currently have conflicting behaviors. When Distraction Free Mode is enabled, users can still zoom out and add patterns, which undermines the purpose of Distraction Free Mode.

This pull request addresses the issue by resolving these conflicting behaviors to maintain the intended functionality of Distraction Free Mode.

The following are the fixes made in this PR:
1. If Distraction Free mode is already activated and the user tries to activate the zoomed-out state using the keyboard shortcut, this action will then disable the distraction-free mode.
2. Zoom Out Toggle will be completely disabled in Distraction Free Mode.
3. If the Zoom Out state is active and the user decides to enter Distraction Free mode either using the keyboard shortcut or from the options, then the zoom out state will reset and then the Distraction Free mode will be activated.

## Why?

The PR ensures Distraction Free Mode functions as intended by preventing disruptive actions like zooming out and adding patterns, preserving focus, and enhancing user experience.

## How?

1. Updated the `toggleDistractionFree()` function to include resetting the zoom level.
2. Disable the Zoom Out Toggle when Distraction Free mode is activated.
3. Update zoom out's callback function to disable distraction-free mode when zoomed out.

## Testing Instructions




1. Go to editing a post.
2. Turn on Distraction Free Mode.
3. The Zoomed Out state Toggle should now not be rendered.
4. If the page is already Zoomed Out, then the page again Zooms In gracefully when distraction-free mode is toggled.

## Screenshots or screencast ( Updated )  



https://github.com/user-attachments/assets/207ec4c9-f224-40e5-beee-a9ad3da3323d
